src/main/java/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/machines/electric/geo/OilPump.java

```diff
@@ -97,28 +97,31 @@ protected void tick(Block b) {
 			}
 		}
 		else {
-			OreGenResource oil = OreGenSystem.getResource("Oil");
-			Chunk chunk = b.getChunk();
-			int supplies = OreGenSystem.getSupplies(oil, chunk, false);
-			
-			if (supplies > 0) {
-				for (int slot : getInputSlots()) {
-					if (SlimefunManager.isItemSimilar(inv.getItemInSlot(slot), new ItemStack(Material.BUCKET), true)) {
+			if (!inv.fits(SlimefunItems.BUCKET_OF_OIL, getOutputSlots())) {
+				return;
+			}
```

Comment:
You could refactor this to simply be

else if (inv.fits(SlimefunItems.BUCKET_OF_OIL, getOutputSlots())) {
// all other code here
}